Allow passing abort/cancel reasons for streams native sources

youennf opened the pull request:
#### 1ff23eb137b9527aa06ed331e81885697cab0033
<pre>
Allow passing abort/cancel reasons for streams native sources
<a href="https://rdar.apple.com/165474756">rdar://165474756</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=303169">https://bugs.webkit.org/show_bug.cgi?id=303169</a>

Reviewed by Alex Christensen.

We pass the reasons for cancel to native ReadableStream sources as this can be used, for instance by WebTransport.
We also fix the abort case for WritableStream since sources were exposing error while they should expose abort.

Test: http/wpt/fs/filestream-abort.https.html
Canonical link: <a href="https://commits.webkit.org/303738@main">https://commits.webkit.org/303738@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dfd7041c70113f3bb22d8572b232065524596cb2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132947 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5448 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44039 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140482 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84978 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/15295baf-6cc8-483e-b739-d90991914dc3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134817 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5865 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5311 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101660 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68991 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4893285c-0d74-4b1e-87a8-634b806fff1e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135893 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4119 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119115 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82460 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f15189c4-0730-4327-88ce-37473b9c05b0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4014 "Passed tests") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1624 "Found 1 new API test failure: TestWebKitAPI.ObscuredContentInsets.TopOverhangColorExtensionLayer (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83715 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113048 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37235 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143135 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5117 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37814 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110037 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5199 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4382 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110218 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28048 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3925 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115377 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58685 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5171 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33731 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5011 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68623 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5261 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5129 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/Modules/streams/WritableStreamSink.idl:
```diff
@@ -30,5 +30,5 @@
 ] interface WritableStreamSink {
     [CallWith=CurrentScriptExecutionContext] Promise<undefined> write(any value);
     undefined close();
-    undefined error(DOMString message);
```
Is removing this an issue?

youennf replied:
No, we should have used `abort` from the start, but we did not have native sinks caring about aborting until WebTransport.

youennf committed:
Allow passing abort/cancel reasons for streams native sources
rdar://165474756
https://bugs.webkit.org/show_bug.cgi?id=303169

Reviewed by Alex Christensen.

We pass the reasons for cancel to native ReadableStream sources as this can be used, for instance by WebTransport.
We also fix the abort case for WritableStream since sources were exposing error while they should expose abort.

Test: http/wpt/fs/filestream-abort.https.html
Canonical link: https://commits.webkit.org/303738@main
